tests/test_cases/test_multi_trig_multi_coro/test_multi_trig_multi_coro.py

```diff
@@ -0,0 +1,58 @@
+import cocotb
+from cocotb.log import SimLog
+from cocotb.triggers import Timer, RisingEdge, FallingEdge
+from cocotb.result import TestFailure, TestSuccess
+
+import sys
+
+@cocotb.coroutine
+def clock_gen(signal, num):
+    for x in range(num):
+        signal <= 0
+        yield Timer(500)
+        signal <= 1
+        yield Timer(500)
+
+class DualMonitor:
+    def __init__(self, signal):
+        self.log = SimLog("cocotb.%s" % (signal._name))
+        self.monitor_edges = [0, 0]
+        self.signal = signal
+
+    @cocotb.coroutine
+    def signal_mon(self, signal, idx):
+        while True:
+            result = yield RisingEdge(signal)
+            self.log.info("signal_mon called on %s" % result)
+            self.monitor_edges[idx] += 1
```

Comment:
I think this test  would be simpler if you dropped the `idx` argument, and just had two counters, `self.mon_edges` and `self.mon_both_edges`